### PR TITLE
New teamtrial theme / URL (unmasking with appshell)

### PR DESCRIPTION
**Pull request by alexnaish**
https://trello.com/c/Uq08XpxR/635-b2b-link-for-unmasking-form

 🐿 v2.10.1

### CODE REVIEW

File: server/modules/form/router.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ import middleware from './middleware';
 
 const router = express.Router();
 
-router.use([middleware.setLocals, middleware.noCache])
+router.use([middleware.noCache]);
```
Comment (remybach) [inline server/modules/form/router.js]: I'm curious... why did you remove it from here given it's used in every route?

File: server/modules/form/middleware.js [hunk 2]
```diff
@@ -16,14 +16,34 @@ export default {
 	},
 
 	setLocals: (req, res, next) => {
-		res.locals.marketingName = req.query.marketingName && escape(req.query.marketingName) || FALLBACK_MARKETING_NAME;
+		res.locals.marketingName = escape(req.params.marketingName || req.query.marketingName || '') || FALLBACK_MARKETING_NAME;
```
Comment (liamkeaton) [inline server/modules/form/middleware.js]: Instead of `''` you could put the `FALLBACK_MARKETING_NAME` in the escape?

Comment (alexnaish) [reply server/modules/form/middleware.js]: I had that originally but then it escapes the spaces and changes the value. Just worried that something somewhere is directing leads based off "propensity organic form" and wont work for "propensity%20organic%20form"

File: test/server/modules/form/api.spec.js [hunk 3]
```diff
@@ -310,7 +362,7 @@ describe('Form', () => {
 
 		});
 
-		it('should correctly map consent fields to Marketo fields', () => {
+		it('should correctly map consent fields to Marketo fields', (done) => {
```
Comment (liamkeaton) [inline test/server/modules/form/api.spec.js]: Ouch

File: test/server/modules/form/api.spec.js [hunk 4]
```diff
@@ -82,7 +94,32 @@ describe('Form', () => {
...
 		});
 
+		it('should customise form for teamtrial theme', () => {
+			return tester.get({ path: '/form/teamtrial' })
+				.then(({ status, headers, $ }) => {
+					expect(status).to.equal(200);
+					expect(headers['cache-control']).to.equal('max-age=0, no-cache, must-revalidate');
+					expect(headers['surrogate-control']).to.equal('max-age=0, no-cache, must-revalidate');
+
+					expect($('header#site-navigation').length).to.equal(1);
```
Comment (liamkeaton) [inline test/server/modules/form/api.spec.js]: Should we add these to the others too, to make sure the header footer is turned off?

File: server/modules/form/router.js (hunk 1 above)
Comment (alexnaish) [reply server/modules/form/router.js]: Due to the way express runs its middleware chains, it was running before the route was matched - meaning it didn't have access to the path and hence no path parameters. It now has to check the path params for the `marketingName` too so it needs to on a "matching" route.

File: server/modules/form/middleware.js [hunk 5]
```diff
@@ -16,14 +16,34 @@ export default {
...
 		res.locals.contentUuid = req.query['ft-content-uuid'] && escape(req.query['ft-content-uuid']) || '';
 		res.locals.sessionToken = req.cookies['FTSession_s'] || '';
 		res.locals.spoorId = req.cookies['spoor-id'] || '';
 		return next();
 	},
 
+	setTemplate: (req, res, next) => {
+
+		const isTeamTrial = res.locals.marketingName === 'teamtrial';
+		res.locals.layout = isTeamTrial ? 'wrapper' : 'vanilla',
```
[the remark this answers is not in the document]
Comment (alexnaish) [reply server/modules/form/middleware.js]: Agreed but didn't want to over-engineer it its only really throwing it into a switch or similar.
